### ISSUE TITLE
Better Asyncio Support

### ISSUE THREAD

**Issue by dummerbd** [pyserial/pyserial#37]
I noticed there's an `aio.py` module in this project that uses the nonblocking mode of the serial connection in order to support use with asyncio. I tried using this on my Mac and my desktop (running Ubuntu 14.04) and got mixed results. For some reason, on my Mac it could only read in a single garbage character `b'0xfe'` or `b'0xff'` and it didn't seem to write correctly. That may or may not be an issue with the `aio.py` code but might have been some other problem.

Either way, I didn't like that the nonblocking mode doesn't support windows so I wrote my own asyncio adapter layer over pyserial using a separate worker thread to handle the blocking serial calls. You can see the code in this gist: https://gist.github.com/dummerbd/75316ab8c109b5bbc46b

If my code looks like a good addition for this library, I can clean it up a bit (like add some logging statements) and make a pull request. If not, let me know what you don't like!

**Comment by zsquareplusc** [comment]
My guess is that Windows support with no threads could be done, looking at the event loop code from asyncio and glancing at some docs on MSDN. However it may be a bit complicated...

I like the workaround using a thread. The [`serial.threaded`](https://github.com/pyserial/pyserial/blob/master/serial/threaded/__init__.py) module already provides a reader thread implementation which may be used for that.

I'd still like to keep a true non-blocking variant, at least on Posix, as it may be useful for small devices running Linux. So it might end up with two implementations.

(re problems on OS X: I'm not sure if the PollSelector will work, at least a few years back when I tried poll instead of select for pySerial, I had to find out that poll was only partially supported by that platform)
